service/interpreter/activityImpl.go

```diff
@@ -54,11 +57,28 @@ func StateApiWaitUntil(ctx context.Context, backendType service.BackendType, inp
 }
 
 // StateDecide is deprecated. Will be removed in next release
-func StateDecide(ctx context.Context, backendType service.BackendType, input service.StateDecideActivityInput) (*iwfidl.WorkflowStateDecideResponse, error) {
-	return StateApiExecute(ctx, backendType, input)
+func StateDecide(ctx context.Context, backendType service.BackendType, input service.StateDecideActivityInput, shouldSendSignalOnCompletion bool) (*iwfidl.WorkflowStateDecideResponse, error) {
+	return StateApiExecute(ctx, backendType, input, shouldSendSignalOnCompletion)
 }
 
-func StateApiExecute(ctx context.Context, backendType service.BackendType, input service.StateDecideActivityInput) (*iwfidl.WorkflowStateDecideResponse, error) {
+func StateApiExecute(ctx context.Context, backendType service.BackendType, input service.StateDecideActivityInput, shouldSendSignalOnCompletion bool) (*iwfidl.WorkflowStateDecideResponse, error) {
+	defer func() {
+		if shouldSendSignalOnCompletion {
+			unifiedCleint := env.GetUnifiedClient()
+			err := unifiedCleint.SignalWithStartWaitForStateCompletionWorkflow(
+				ctx, client.StartWorkflowOptions{
+					ID:                       service.IwfSystemConstPrefix + input.Request.Context.WorkflowId + "_" + *input.Request.Context.StateExecutionId,
+					TaskQueue:                env.GetTaskQueue(),
+					WorkflowExecutionTimeout: 600 * time.Second, // TODO: make it configurable
```

Comment:
This can be passed from the InterpreterWorkflow, they should use the same timeout